packages/wdio-sumologic-reporter/src/index.js

```diff
@@ -1 +1,136 @@
+
+        if (typeof this.options.sourceAddress !== 'string') {
+            log.error('Sumo Logic requires "sourceAddress" paramater')
+        }
+
+        // Cache of entries we are yet to sync
+        this.unsynced = []
+        this.inSync = false
+
+        this.errorCount = 0
+        this.specs = {}
+        this.results = {}
+        this.interval = setInterval(::this.sync, this.options.syncInterval)
+    }
+
+    onRunnerStart (runner) {
+        this.unsynced.push(stringify({
+            time: dateFormat(new Date(), DATE_FORMAT),
+            event: 'runner:start',
+            data: runner
+        }))
+    }
+
+    onSuiteStart (suite) {
+        this.unsynced.push(stringify({
+            time: dateFormat(new Date(), DATE_FORMAT),
+            event: 'suite:start',
+            data: suite
+        }))
+    }
+
+    onTestStart (test) {
+        this.unsynced.push(stringify({
+            time: dateFormat(new Date(), DATE_FORMAT),
+            event: 'test:start',
+            data: test
+        }))
+    }
+
+    onTestSkip (test) {
+        this.unsynced.push(stringify({
+            time: dateFormat(new Date(), DATE_FORMAT),
+            event: 'test:skip',
+            data: test
+        }))
+    }
+
+    onTestPass (test) {
+        this.unsynced.push(stringify({
+            time: dateFormat(new Date(), DATE_FORMAT),
+            event: 'test:pass',
+            data: test
+        }))
+    }
+
+    onTestFail (test) {
+        this.unsynced.push(stringify({
+            time: dateFormat(new Date(), DATE_FORMAT),
+            event: 'test:fail',
+            data: test
+        }))
+    }
+
+    onTestEnd (test) {
+        this.unsynced.push(stringify({
+            time: dateFormat(new Date(), DATE_FORMAT),
+            event: 'test:end',
+            data: test
+        }))
+    }
+
+    onSuiteEnd (suite) {
+        this.unsynced.push(stringify({
+            time: dateFormat(new Date(), DATE_FORMAT),
+            event: 'suite:end',
+            data: suite
+        }))
+    }
+
+    onRunnerEnd (runner) {
+        this.unsynced.push(stringify({
+            time: dateFormat(new Date(), DATE_FORMAT),
+            event: 'runner:end',
+            data: runner
+        }))
+    }
+
+    sync () {
+        if (this.inSync || this.unsynced.length === 0 || typeof this.options.sourceAddress !== 'string') {
+            return
+        }
+
+        const logLines = this.unsynced.slice(0, MAX_LINES).join('\n')
+        this.inSync = true
+
+        request({
+            method: 'POST',
+            uri: this.options.sourceAddress,
+            body: logLines
+        }, (err, resp) => {
+            const failed = Boolean(err) || resp.status < 200 || resp.status >= 400
+
+            if (failed) {
+                log.error('failed send data to Sumo Logic:\n', err.stack ? err.stack : err)
+            } else {
+                this.unsynced.splice(0, MAX_LINES)
+            }
+
+            this.inSync = false
```

Comment:
should inSync be false when there is no error? Or does log.error return instantly in which case this code will never be reached?

Reply:
ah no .. `inSync` means whether or not we are currently sending out a request and waiting for a response. I think I can rename it to something more suiteable